neo4j/internal/pool/server.go

```diff
@@ -41,17 +41,21 @@ var sharedRoundRobin uint32
 
 const rememberFailedConnectDuration = 3 * time.Minute
 
-// Returns a idle connection if any
-func (s *server) getIdle() db.Connection {
+// Returns an idle connection if any
+func (s *server) getIdle(ctx context.Context, idlenessThreshold time.Duration) db.Connection {
```

Comment:
Do the looping over idle connections, waiting for the result of the forced reset. https://neo4j.slack.com/archives/D01MXVBHT26/p1651766602894969